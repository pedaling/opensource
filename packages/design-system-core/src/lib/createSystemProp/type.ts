import type { Theme } from '../types';

export type InterpolationFn = (props: Record<string, any>) => Record<string, any>;

export type SystemPropConfig = {
  property: string;
  styleProperty?: string;
  scale?: keyof Pick<Theme, 'colors' | 'space' | 'opacity'>;
  disabled?: boolean;
  transform?: (value: any) => Record<string, any> | undefined;
};

export type SystemProp = {
  (props: Record<string, any>): Record<string, any>;
  propName: string;
  disabled: boolean;
};
