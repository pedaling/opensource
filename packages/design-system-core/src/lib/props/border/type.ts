import type { ResponsiveValue } from '../../types';

export type BorderProps = {
  borderWidth?: ResponsiveValue<number>;
  borderStyle?: ResponsiveValue<'solid' | 'none'>;
  borderColor?: ResponsiveValue<string>;
  borderTopWidth?: ResponsiveValue<number>;
  borderTopStyle?: ResponsiveValue<'solid' | 'none'>;
  borderTopColor?: ResponsiveValue<string>;
  borderRightWidth?: ResponsiveValue<number>;
  borderRightStyle?: ResponsiveValue<'solid' | 'none'>;
  borderRightColor?: ResponsiveValue<string>;
  borderBottomWidth?: ResponsiveValue<number>;
  borderBottomStyle?: ResponsiveValue<'solid' | 'none'>;
  borderBottomColor?: ResponsiveValue<string>;
  borderLeftWidth?: ResponsiveValue<number>;
  borderLeftStyle?: ResponsiveValue<'solid' | 'none'>;
  borderLeftColor?: ResponsiveValue<string>;
  borderRadius?: ResponsiveValue<number | string>;
  borderTopLeftRadius?: ResponsiveValue<number | string>;
  borderTopRightRadius?: ResponsiveValue<number | string>;
  borderBottomLeftRadius?: ResponsiveValue<number | string>;
  borderBottomRightRadius?: ResponsiveValue<number | string>;
  outlineWidth?: ResponsiveValue<number>;
  outlineStyle?: ResponsiveValue<'solid' | 'none'>;
  outlineColor?: ResponsiveValue<string>;
  outlineOffset?: ResponsiveValue<number>;
};
