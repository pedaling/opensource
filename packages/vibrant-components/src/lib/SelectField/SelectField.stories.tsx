import type { ComponentStory, ComponentMeta } from '@storybook/react';
import { Box } from '@vibrant-ui/core';
import { Body } from '../Body';
import { HStack } from '../HStack';
import { VStack } from '../VStack';
import { SelectField } from './SelectField';

export default {
  title: 'SelectField',
  component: SelectField,
  args: {
    label: 'label',
    options: [
      {
        label: 'option 1',
        value: '1',
      },
      {
        label: 'option 2',
        value: '2',
      },
      {
        label: 'option 3',
        value: '3',
      },
      {
        label: 'option 4',
        value: '4',
      },
      {
        label: 'option 5',
        value: '5',
      },
      {
        label: 'option 6',
        value: '6',
      },
      {
        label: 'option 7',
        value: '7',
      },
      {
        label: 'option 8',
        value: '8',
      },
    ],
    inlineLabel: true,
  },
} as ComponentMeta<typeof SelectField>;

export const Basic: ComponentStory<typeof SelectField> = props => (
  <Box>
    <Box height={500} />
    <SelectField {...props} />
    <Box height={500} />
  </Box>
);

export const WithRenderItem: ComponentStory<typeof SelectField> = props => (
  <Box>
    <Box height={500} />
    <SelectField
      {...props}
      renderItem={({ label }) => (
        <VStack spacing={8}>
          <Body level={2} color="onView1">
            {label}
          </Body>
          <HStack spacing={4}>
            <Body level={2} weight="bold" color="error">
              22%
            </Body>
            <Body level={2} weight="bold" color="onView1">
              224,700Ïõê
            </Body>
          </HStack>
        </VStack>
      )}
    />
    <Box height={500} />
  </Box>
);
