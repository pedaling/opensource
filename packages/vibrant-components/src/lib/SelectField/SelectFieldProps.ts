import { withVariation } from '@vibrant-ui/core';
import type { SelectOptionGroupProps } from '../SelectOptionGroup';

export type SelectFieldProps = Pick<SelectOptionGroupProps, 'options' | 'renderItem'> & {
  state?: 'default' | 'error';
  errorMessage?: string;
} & (
    | {
        label: string;
        inlineLabel?: boolean;
        placeholder?: never;
      }
    | {
        label?: never;
        inlineLabel?: never;
        placeholder: string;
      }
  );

export const withSelectFieldVariation = withVariation<SelectFieldProps>()();
