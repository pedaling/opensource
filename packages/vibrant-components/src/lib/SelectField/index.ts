export { SelectField } from './SelectField';
export type { SelectFieldProps } from './SelectFieldProps';
