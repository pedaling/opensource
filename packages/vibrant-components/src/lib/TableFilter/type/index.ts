export type StringFilterOperator = 'contains' | 'empty' | 'equals' | 'notContains' | 'notEmpty' | 'notEquals';

export type StringFilter = {
  value: string;
  operator: StringFilterOperator[];
};

export type Filter = StringFilter & { dataKey: string };
