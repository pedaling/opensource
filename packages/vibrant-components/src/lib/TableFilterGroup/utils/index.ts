import type { DateFilterOperator, MultiSelectFilterOperator, Option, StringFilterOperator } from '../types';

export const isValueRequiredOperator = (op: string) => op === 'empty' || op === 'notEmpty';

export const isStringFilterValid = (filter: { value: string; operator: StringFilterOperator }) =>
  Boolean(filter.value) || isValueRequiredOperator(filter.operator);

export const isDateFilterValid = (filter: { value: Date[]; operator: DateFilterOperator }) => {
  if (isValueRequiredOperator(filter.operator)) return true;

  if (filter.operator === 'between') return filter.value.length >= 2;

  return filter.value.length >= 1;
};

export const isMultiSelectFilterValid = (filter: { value: Option['value'][]; operator: MultiSelectFilterOperator }) =>
  Boolean(filter.value.length > 0) || isValueRequiredOperator(filter.operator);
