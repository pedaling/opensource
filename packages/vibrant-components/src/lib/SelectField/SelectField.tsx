import { useEffect, useMemo, useRef, useState } from 'react';
import type { TextInputRef } from '@vibrant-ui/core';
import { Box, PressableBox, TextInput, getElementPosition } from '@vibrant-ui/core';
import { Icon } from '@vibrant-ui/icons';
import { Body } from '../Body';
import { Dismissible } from '../Dismissible';
import { HStack } from '../HStack';
import { SelectOptionGroup } from '../SelectOptionGroup';
import { Space } from '../Space';
import { withSelectFieldVariation } from './SelectFieldProps';

export const SelectField = withSelectFieldVariation(
  ({
    label,
    placeholder,
    inlineLabel,
    options,
    state: stateProp = 'default',
    helperText,
    renderOption,
    disabled,
    onValueChange,
    ...restProps
  }) => {
    const [state, setState] = useState<'default' | 'error'>(stateProp);
    const [isFocused, setIsFocused] = useState(false);
    const [isOpened, setIsOpened] = useState(false);
    const [direction, setDirection] = useState<'down' | 'up'>('down');
    const [focusIndex, setFocusIndex] = useState(-1);
    const [optionGroupMaxHeight, setOptionGroupMaxHeight] = useState<number | string>('auto');
    const [selectedOptionIndex, setSelectedOptionIndex] = useState<number>(-1);

    const ref = useRef<HTMLElement>(null);
    const inputRef = useRef<TextInputRef>(null);

    const labelColor = useMemo(() => {
      if (disabled) {
        return 'onView3';
      }

      if (state === 'error') {
        return 'error';
      }

      return 'onView2';
    }, [disabled, state]);
    const borderColor = useMemo(() => {
      if (disabled) {
        return 'outline1';
      }

      if (state === 'error') {
        return 'error';
      }

      return isFocused ? 'outlineNeutral' : 'outline1';
    }, [disabled, state, isFocused]);

    const selectedOption = options[selectedOptionIndex];

    useEffect(() => {
      if (!selectedOption) {
        return;
      }

      setState('default');

      onValueChange?.(selectedOption.value);
      // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [selectedOption]);

    useEffect(() => {
      setState(stateProp);
    }, [stateProp]);

    const open = async (index: number) => {
      if (!ref.current || disabled) {
        return;
      }

      inputRef.current?.focus();

      const { top, bottom } = await getElementPosition(ref.current);

      const spaceAbove = top;
      const spaceBelow = bottom;

      if (selectedOptionIndex !== -1) {
        setFocusIndex(selectedOptionIndex);
      } else {
        setFocusIndex(index);
      }

      if (spaceAbove > spaceBelow) {
        setOptionGroupMaxHeight(spaceAbove - 44);

        setDirection('up');
      } else {
        setOptionGroupMaxHeight(spaceBelow - 84);

        setDirection('down');
      }

      setIsFocused(true);

      setIsOpened(true);
    };

    const close = () => {
      setFocusIndex(-1);

      setIsOpened(false);
    };

    return (
      <Box zIndex={1}>
        <TextInput
          ref={inputRef}
          type="text"
          defaultValue={selectedOption?.value ?? ''}
          readOnly={true}
          disabled={disabled}
          hidden={true}
          onSubmit={() => {
            if (!isOpened) {
              open(0);

              return;
            }

            if (focusIndex !== -1) {
              setSelectedOptionIndex(focusIndex);

              close();
            }
          }}
          onKeyPress={({ key, prevent }) => {
            if (key === (direction === 'up' ? 'ArrowDown' : 'ArrowUp')) {
              setFocusIndex(Math.max(0, focusIndex - 1));
            }

            if (key === (direction === 'up' ? 'ArrowUp' : 'ArrowDown')) {
              setFocusIndex(Math.min(options.length - 1, focusIndex + 1));
            }

            prevent();
          }}
        />
        <Box position="relative">
          <Dismissible active={isFocused} onDismiss={() => setIsFocused(false)}>
            <PressableBox
              ref={ref}
              alignItems="center"
              flexDirection="row"
              height={50}
              px={16}
              borderWidth={1}
              borderStyle="solid"
              borderColor={borderColor}
              borderRadius={2}
              cursor={disabled ? 'default' : 'pointer'}
              onClick={() => (isOpened ? close() : open(-1))}
              {...restProps}
            >
              <HStack width="100%">
                <Box as="span" flex={1}>
                  {selectedOption ? (
                    <Box flexDirection={inlineLabel ? 'row' : 'column'}>
                      {Boolean(label) && (
                        <>
                          <Body level={2} color={labelColor} lineLimit={1} wordBreak="break-all" wordWrap="break-word">
                            {label}
                          </Body>
                          <Box as="span" flexShrink={0} hidden={!inlineLabel}>
                            <Body level={2} color={disabled ? 'onView3' : 'onView2'}>
                              &nbsp;/&nbsp;
                            </Body>
                          </Box>
                        </>
                      )}
                      <Body
                        level={2}
                        color={disabled ? 'onView3' : 'onView1'}
                        wordBreak="break-all"
                        wordWrap="break-word"
                        lineLimit={1}
                      >
                        {selectedOption.label}
                      </Body>
                    </Box>
                  ) : (
                    <Body level={2} color={labelColor} lineLimit={1} wordBreak="break-all" wordWrap="break-word">
                      {label || placeholder}
                    </Body>
                  )}
                </Box>
                <Space width={12} />
                <Icon.ArrowTriangleDown.Regular
                  size={20}
                  fill={disabled ? 'onView3' : state === 'error' ? 'error' : 'onView1'}
                />
              </HStack>
            </PressableBox>
          </Dismissible>
          <Dismissible active={isOpened} onDismiss={close}>
            <Box
              position="absolute"
              zIndex={1}
              width="100%"
              maxHeight={[optionGroupMaxHeight, optionGroupMaxHeight, 320]}
              hidden={!isOpened}
              {...(direction === 'down' ? { top: 54 } : { bottom: 54 })}
            >
              <SelectOptionGroup
                onOptionClick={index => {
                  setSelectedOptionIndex(index);

                  close();

                  inputRef.current?.focus();
                }}
                state={state}
                renderOption={renderOption}
                reverse={direction === 'up'}
                options={options}
                focusIndex={focusIndex}
              />
            </Box>
          </Dismissible>
        </Box>
        {Boolean(helperText) && (
          <Body level={4} color={state === 'error' ? 'error' : 'onView2'} wordBreak="keep-all" wordWrap="break-word">
            {helperText}
          </Body>
        )}
      </Box>
    );
  }
);
