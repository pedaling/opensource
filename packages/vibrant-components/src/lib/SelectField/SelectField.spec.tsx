import { waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { PortalRootProvider } from '@vibrant-ui/core';
import type { ReactRenderer } from '@vibrant-ui/utils/testing-web';
import { createReactRenderer } from '@vibrant-ui/utils/testing-web';
import { SelectField } from './SelectField';

describe('<SelectField />', () => {
  const { render } = createReactRenderer();

  let renderer: ReactRenderer;

  describe('when size is lg', () => {
    beforeEach(async () => {
      renderer = render(
        <SelectField
          size="lg"
          label="label"
          options={[
            {
              label: 'option 1',
              value: '1',
            },
            {
              label: 'option 2',
              value: '2',
            },
          ]}
        />
      );
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });

  describe('when size is md', () => {
    beforeEach(async () => {
      renderer = render(
        <SelectField
          size="md"
          label="label"
          options={[
            {
              label: 'option 1',
              value: '1',
            },
            {
              label: 'option 2',
              value: '2',
            },
          ]}
        />
      );
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });

  describe('when size is sm', () => {
    beforeEach(async () => {
      renderer = render(
        <SelectField
          size="sm"
          label="label"
          options={[
            {
              label: 'option 1',
              value: '1',
            },
            {
              label: 'option 2',
              value: '2',
            },
          ]}
        />
      );
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });

  describe('when select option', () => {
    const handleValueChange = jest.fn();

    beforeEach(async () => {
      renderer = render(
        <PortalRootProvider zIndex={1}>
          <SelectField
            size="sm"
            label="label"
            onValueChange={handleValueChange}
            options={[
              {
                label: 'option 1',
                value: '1',
              },
              {
                label: 'option 2',
                value: '2',
              },
            ]}
          />
        </PortalRootProvider>
      );

      const selectField = await renderer.findByTestId('select-field');

      await waitFor(() => userEvent.click(selectField.children[1]));

      const optionItems = await renderer.findAllByTestId('select-option-item');

      await waitFor(() => optionItems[0].click());
    });

    it('onValueChange called', () => {
      expect(handleValueChange).toBeCalledTimes(1);
    });
  });
});
