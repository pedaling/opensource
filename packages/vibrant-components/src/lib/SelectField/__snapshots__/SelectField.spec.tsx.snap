// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectField /> when size is lg match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  background-color: #00000000;
  border-width: 0;
  color: #0c0c0c;
  -webkit-text-fill-color: #0c0c0c;
  text-fill-color: #0c0c0c;
  font-size: 0.875rem;
  line-height: 1.125rem;
  width: 100%;
  padding: 0;
  position: absolute;
  height: 0;
  opacity: 0;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 50px;
  padding-left: 15px;
  padding-right: 15px;
  border-width: 1px;
  border-style: solid;
  border-color: #00000026;
  border-radius: 2px;
  background-color: #ffffff;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding-right: 12px;
  min-width: 0;
}

.emotion-5 {
  text-align: left;
  color: #5b5b5b;
  -webkit-text-fill-color: #5b5b5b;
  text-fill-color: #5b5b5b;
  display: -webkit-box;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 1;
  word-break: break-all;
  word-wrap: break-word;
  font-size: 0.875rem;
  line-height: 1.125rem;
  font-weight: 400;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 20px;
  height: 20px;
  fill: #0c0c0c;
}

<div>
  <div
    class="emotion-0"
    data-testid="select-field"
  >
    <input
      autocomplete="off"
      class="emotion-1"
      inputmode="text"
      readonly=""
      type="text"
      value=""
    />
    <div
      class="emotion-2"
      tabindex="0"
    >
      <div
        class="emotion-3"
      >
        <span
          class="emotion-4"
        >
          <span
            class="emotion-5"
            data-testid="body"
          >
            label
          </span>
        </span>
        <svg
          class="emotion-6"
          data-testid="arrowtriangledown-thin"
          viewBox="0 0 24 24"
        >
          <path
            class="emotion-0"
            d="m7.05 9.4 4.75 7c.1.15.3.15.4 0l4.75-7c.1-.15 0-.4-.2-.4h-9.5c-.2 0-.3.25-.2.4Z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`<SelectField /> when size is md match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  background-color: #00000000;
  border-width: 0;
  color: #0c0c0c;
  -webkit-text-fill-color: #0c0c0c;
  text-fill-color: #0c0c0c;
  font-size: 0.875rem;
  line-height: 1.125rem;
  width: 100%;
  padding: 0;
  position: absolute;
  height: 0;
  opacity: 0;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 38px;
  padding-left: 9px;
  padding-right: 9px;
  border-width: 1px;
  border-style: solid;
  border-color: #00000026;
  border-radius: 2px;
  background-color: #ffffff;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding-right: 8px;
  min-width: 0;
}

.emotion-5 {
  text-align: left;
  color: #5b5b5b;
  -webkit-text-fill-color: #5b5b5b;
  text-fill-color: #5b5b5b;
  display: -webkit-box;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 1;
  word-break: break-all;
  word-wrap: break-word;
  font-size: 0.875rem;
  line-height: 1.125rem;
  font-weight: 400;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 20px;
  height: 20px;
  fill: #0c0c0c;
}

<div>
  <div
    class="emotion-0"
    data-testid="select-field"
  >
    <input
      autocomplete="off"
      class="emotion-1"
      inputmode="text"
      readonly=""
      type="text"
      value=""
    />
    <div
      class="emotion-2"
      tabindex="0"
    >
      <div
        class="emotion-3"
      >
        <span
          class="emotion-4"
        >
          <span
            class="emotion-5"
            data-testid="body"
          >
            label
          </span>
        </span>
        <svg
          class="emotion-6"
          data-testid="arrowtriangledown-thin"
          viewBox="0 0 24 24"
        >
          <path
            class="emotion-0"
            d="m7.05 9.4 4.75 7c.1.15.3.15.4 0l4.75-7c.1-.15 0-.4-.2-.4h-9.5c-.2 0-.3.25-.2.4Z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`<SelectField /> when size is sm match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  background-color: #00000000;
  border-width: 0;
  color: #0c0c0c;
  -webkit-text-fill-color: #0c0c0c;
  text-fill-color: #0c0c0c;
  font-size: 0.875rem;
  line-height: 1.125rem;
  width: 100%;
  padding: 0;
  position: absolute;
  height: 0;
  opacity: 0;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 30px;
  padding-left: 7px;
  padding-right: 7px;
  border-width: 1px;
  border-style: solid;
  border-color: #00000026;
  border-radius: 2px;
  background-color: #ffffff;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding-right: 6px;
  min-width: 0;
}

.emotion-5 {
  text-align: left;
  color: #5b5b5b;
  -webkit-text-fill-color: #5b5b5b;
  text-fill-color: #5b5b5b;
  display: -webkit-box;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 1;
  word-break: break-all;
  word-wrap: break-word;
  font-size: 0.75rem;
  line-height: 1rem;
  font-weight: 400;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 16px;
  height: 16px;
  fill: #0c0c0c;
}

<div>
  <div
    class="emotion-0"
    data-testid="select-field"
  >
    <input
      autocomplete="off"
      class="emotion-1"
      inputmode="text"
      readonly=""
      type="text"
      value=""
    />
    <div
      class="emotion-2"
      tabindex="0"
    >
      <div
        class="emotion-3"
      >
        <span
          class="emotion-4"
        >
          <span
            class="emotion-5"
            data-testid="body"
          >
            label
          </span>
        </span>
        <svg
          class="emotion-6"
          data-testid="arrowtriangledown-thin"
          viewBox="0 0 24 24"
        >
          <path
            class="emotion-0"
            d="m7.05 9.4 4.75 7c.1.15.3.15.4 0l4.75-7c.1-.15 0-.4-.2-.4h-9.5c-.2 0-.3.25-.2.4Z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
